ddl/schematracker/checker.go

```diff
@@ -62,13 +63,32 @@ func (d *Checker) Enable() {
 	d.closed = false
 }
 
+// CreateTestDB creates a `test` database like the default behaviour of TiDB.
+func (d Checker) CreateTestDB() {
+	d.tracker.createTestDB()
+}
+
 func (d Checker) checkDBInfo(ctx sessionctx.Context, dbName model.CIStr) {
 	if d.closed {
 		return
 	}
 	dbInfo, _ := d.realDDL.GetInfoSchemaWithInterceptor(ctx).SchemaByName(dbName)
 	dbInfo2 := d.tracker.SchemaByName(dbName)
 
+	if dbInfo == nil || dbInfo2 == nil {
+		if dbInfo == nil && dbInfo2 == nil {
+			return
+		}
+		if dbInfo == nil {
+			errStr := fmt.Sprintf("%s should not exist", dbName)
+			panic(errStr)
+		}
+		if dbInfo2 == nil {
+			errStr := fmt.Sprintf("%s should exist", dbName)
+			panic(errStr)
```

Comment:
Maybe we could combine those lines like 
```
errStr := fmt.Sprintf("inconsistent dbInfo, dbName: %s, real ddl: %p, schematracker：%p", dbName, dbInfo, dbInfo2)
panic(errStr)
```

I think it's better.

Reply:
will do it tomorrow

Reply:
https://github.com/pingcap/tidb/pull/36077/commits/b57935ebae48c6ffbb9579d0586fe956fc22883c